crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -295,6 +295,8 @@ where
             .await
             .map_err(EnvironmentError2::EvalCatalog)?;
 
+        std::fs::write(self.path.join("catalog.json"), catalog_value.to_string())
```

Comment:
This needs to be in `pkgs/default/catalog.json`. You might want to grab https://github.com/flox/flox/pull/230/files#diff-1fda1bcccf425544df5f1ab9ff436f3053715f85b04157f52fd5de3f4f88b7a6R432